src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-details/pool-details.component.spec.ts

```diff
@@ -15,6 +15,16 @@ describe('PoolDetailsComponent', () => {
   let poolDetailsComponent: PoolDetailsComponent;
   let fixture: ComponentFixture<PoolDetailsComponent>;
 
+  const updatePoolSelection = (selection: any) => {
+    poolDetailsComponent.selection = selection;
+    poolDetailsComponent.ngOnChanges();
```

Comment:
I have checked a few links about triggering `ngOnChanges` manually ([[1]](https://stackoverflow.com/questions/44941635/how-to-trigger-a-change-event-manually-angular2), [[2]](https://medium.com/@isaacplmann/ngonchanges-only-runs-when-the-input-change-comes-from-a-template-binding-like-component-8797b759ba0b), [[3]](https://www.bennadel.com/blog/3057-manually-triggering-ngonchanges-inside-an-ngmodel-value-accessor-in-angular-2-beta-11.htm)) and they all involve that directly calling `ngOnChanges` is a smell of wrong code.

Reply:
I agree with the provided links that it's a smell of code if we trigger `ngOnChanges` manually in a component like the above but this is a test case to simulate what would happen if the selection were changed by the outside which would trigger `ngOnChanges` because the input has changed.

I'm not sure if writing a mock component that uses the pools details component is very useful as the component is only used once by the pool list component, that's why the change of the selection is mocked here.

Reply:
> I agree with the provided links that it's a smell of code if we trigger ngOnChanges manually in a component like the above but this is a test case to simulate what would happen if the selection were changed by the outside which would trigger ngOnChanges because the input has changed.

You're completely right! Didn't realized that this was a unit test...